Sources/NIOHTTP2/HTTP2Settings.swift

```
@@ -12,101 +12,232 @@
+        case .enablePush(let b), .acceptCacheDigest(let b), .enableConnectProtocol(let b):
+            return Int(self.identifier) & 31 + (b ? 1 : 0)
+        }
     }
+}
 
-    /// Create a new `HTTP2Setting` from nghttp2's raw representation.
-    internal init(fromNghttp2 setting: nghttp2_settings_entry) {
-        self.parameter = HTTP2SettingsParameter(fromNetwork: setting.settings_id)
-        self._value = setting.value
+internal extension nghttp2_settings_entry {
+    internal init(nioSetting: HTTP2Setting) {
+        switch nioSetting {
+        case .headerTableSize(let v):
+            self.init(settings_id: Int32(NGHTTP2_SETTINGS_HEADER_TABLE_SIZE.rawValue), value: UInt32(v))
+        case .enablePush(let v):
+            self.init(settings_id: Int32(NGHTTP2_SETTINGS_ENABLE_PUSH.rawValue), value: v ? 1 : 0)
+        case .maxConcurrentStreams(let v):
+            self.init(settings_id: Int32(NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS.rawValue), value: UInt32(v))
+        case .initialWindowSize(let v):
+            self.init(settings_id: Int32(NGHTTP2_SETTINGS_INITIAL_WINDOW_SIZE.rawValue), value: UInt32(v))
+        case .maxFrameSize(let v):
+            self.init(settings_id: Int32(NGHTTP2_SETTINGS_MAX_FRAME_SIZE.rawValue), value: UInt32(v))
+        case .maxHeaderListSize(let v):
+            self.init(settings_id: Int32(NGHTTP2_SETTINGS_MAX_HEADER_LIST_SIZE.rawValue), value: UInt32(v))
+        case .acceptCacheDigest(let v):
+            self.init(settings_id: 0x7, value: v ? 1 : 0)
+        case .enableConnectProtocol(let v):
+            self.init(settings_id: 0x8, value: v ? 1 : 0)
+        }
     }
 }
 
-extension HTTP2Setting: Equatable {
-    public static func ==(lhs: HTTP2Setting, rhs: HTTP2Setting) -> Bool {
-        return lhs.parameter == rhs.parameter && lhs._value == rhs._value
+extension HTTP2Setting {
+    // nullable *and* throws? Invalid data causes an error, but unknown setting types return 'nil' quietly.
```

We should probably have an escape hatch for unknown settings types, rather than returning `nil` and throwing them away.

---

With the non-enum design, that won't be a problem any more: I'll have an 'unknown setting' type to return.